### PR TITLE
Estimate LLM inference impacts from Azure Foundry token meters

### LINKED ISSUE #264: Add EcoLogits estimates for Azure OpenAI
The EcoLogits lookup is provider-neutral but currently only Bedrock has a billing adapter.

Azure OpenAI usage is billed through token meters under Azure OpenAI / Foundry Models, with names such as:

```
GPT 5 outpt Glbl 1M Tokens
5.1 codex opt Gl 1M Tokens
5 mini pp Inp Gl 1M Tokens
```

A similar Azure module could:

1. bind the native Azure and FOCUS meter, quantity and unit columns
2. recognise supported OpenAI token meters and extract the model and input/output direction
3. add the Azure aliases to `mapping.csv` and reuse existing coefficients where available
4. normalise the 1K/1M units and estimate output-token rows only
5. enable the module in both Azure default configurations

This would be a first step towards the `ecologits.azure` split sketched in #143.

PTU/hourly meters, fine-tuning and non-OpenAI Foundry models are out of scope initially.

Happy to work on this. @jnioche, wdyt?

### PR DESCRIPTION
**Pull request by dpol1**
Fixes #264

The EcoLogits lookup is provider-neutral but only Bedrock had a billing adapter, so LLM usage on Azure stayed unestimated. This adds `AzureFoundryTokenEcoLogits`, the Azure counterpart of `BedrockEcoLogits`: it recognises the token meters billed under the `Azure OpenAI` / `Foundry Models` category, extracts the model label and the token direction from `MeterName`, maps the label to the EcoLogits coefficients via `mapping.csv` and estimates energy and embodied emissions for the output-token rows. The rows then flow through PWUE, carbon intensity, operational emissions and water like any other usage. First step towards the `ecologits.azure` split sketched in #143.

One thing worth calling out: `Quantity` (and `ConsumedQuantity` in FOCUS reports) already holds the consumed token count. The `1K`/`1M` in `UnitOfMeasure` only describes the pricing block — Microsoft's FOCUS conversion rules define `ConsumedQuantity = Quantity` and `ContractedCost = UnitPrice × Quantity / x_PricingBlockSize` — so, unlike the Bedrock adapter with `pricing_unit`, no unit normalisation is applied. The issue text ("normalise the 1K/1M units") predates that finding.

## Changes

- **`ecologits/AzureFoundryTokenEcoLogits`**: filters on the meter category, parses `MeterName` (input markers `Inp/inpt/input`, output markers `Outp/outpt/opt/out/output`; deployment and pricing qualifiers such as `Batch`, `cchd`, `pp`, `ShortCo` are dropped from the label), scores output tokens only and leaves everything else untouched: unknown labels (warned once by `EcoLogits`), input and cached tokens, meters without a direction marker (reasoning, embeddings), non-token meters. Native and FOCUS bindings (`MeterCategory`/`MeterName`/`Quantity` ↔ `x_SkuMeterCategory`/`x_SkuMeterName`/`ConsumedQuantity`).
- **`ecologits/mapping.csv`**: ten Azure OpenAI labels (`gpt 5`, `5`, `5 mini`, `5 nano`, `5.1`, `5.1 codex`, `5.4`, `gpt 4.1`, `o1`, `o1 1217`), each checked against the bundled `coefficients.csv`. Meter names taken from the Azure Retail Prices API.
- **Default configurations**: the module is enabled in `default-config-azure.json` and `default-config-azure-focus.json`, before `PWUE` as Bedrock is in the AWS one.
- **Tests**: `AzureFoundryTokenEcoLogitsTest` (parser cases use real meter names, e.g. `GPT 5 outpt Glbl 1M Tokens`, `5.1 codex opt Gl 1M Tokens`, `5.6 terra ShortCo Cd Inp PP Gl 1M Tokens`; native/FOCUS invariant: the same inference in both formats yields the same impacts) and `AzureFoundryTokenEndToEndTest`, which runs two minimal exports through the real default configurations and the `EnrichmentPipeline`.
- **Docs**: module reference.

## Verification

350 tests pass. Also ran the job in the Docker image with a fresh jar on the native fixture:

| MeterName | Quantity | kWh | embodied gCO2eq | operational gCO2eq |
|---|---|---|---|---|
| `GPT 5 outpt Glbl 1M Tokens` | 1,000,000 | 2.015 | 52.7 | 92.6 |
| `gpt 4.1 Outp regnl Tokens` | 2,000 | 0.0045 | 0.13 | 0.21 |
| input / unmapped model / non-token / non-usage rows | | null | null | null |

No file overlaps with #266, so the two can be merged in either order.

## Left out deliberately

- **PTU, hourly hosting and fine-tuning meters**: not token-based.
- **Reasoning and embedding meters**: no input/output marker, so they are skipped — a slight underestimate for reasoning models until the fragment is confirmed on real data.
- **Other Foundry model families**: the Retail Prices catalogue shows the same meter contract for Llama, Mistral, Cohere, Grok, Qwen (`3.3 70b Outp FT DZ Tokens`, `4.3 Outp Glbl L Tokens`, unit `1K`), so they only need `mapping.csv` entries verified against real exports; DeepSeek, Grok and Qwen have no EcoLogits coefficients anyway. Separate issue once this lands.
- **`gpt-5.4 pro/nano` and `gpt-5.5` aliases**: need the refreshed coefficients from #266; three lines once both are merged. `gpt-5.6` is not in the EcoLogits API yet.
- **Input-token energy**: same choice as the Bedrock adapter (EcoLogits attributes the generation cost to output tokens, see mlco2/ecologits#86).

## A real export would help

The parser is validated against the Retail Prices catalogue and synthetic rows, not against an actual EA/MCA or FOCUS export. If anyone has Azure OpenAI usage, a redacted tuple is enough to confirm the contract — export type plus `MeterCategory`, `MeterName`, `UnitOfMeasure`, `Quantity` (no cost or identifiers).

### CODE REVIEW

**Review by jnioche — APPROVED** [review]
thanks @dpol1
